go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -822,11 +855,24 @@ func (be *BuiltinBackupEngine) backupFile(ctx context.Context, params BackupPara
 			compressor, err = newBuiltinCompressor(CompressionEngineName, writer, params.Logger)
 		}
 		if err != nil {
-			return vterrors.Wrap(err, "can't create compressor")
+			finalErr = vterrors.Wrap(err, "can't create compressor")
+			return finalErr
```

Comment:
You can return the error directly, you don't have to assign to finallErr, that happens implicitly.

Reply:
OMG TIL

Reply:
OK fixed, basically reverting much into the original assignment logic.